packages/quantic/force-app/main/default/lwc/quanticHeadlessLoader/quanticHeadlessLoader.js

```diff
@@ -166,8 +167,8 @@ function setComponentInitialized(element, engineId) {
     throw new Error('Fatal Error: Component was not registered before initialization');
   }
   component.initialized = true;
-  if (areAllComponentsInitialized(engineId)) {
-    debounceInitialSearch(engineId);
+  if (window.coveoHeadless[engineId].initializedCallback && areAllComponentsInitialized(engineId)) {
+    debounceInitializedCallback(engineId);
   }
 }
 
```

Comment:
I think the `await` is missing before `initEngine(engineId)` in this function.

    /**
     * Returns headless engine promise.
     * @param {string} engineId The id of the engine.
     */
    async function getHeadlessEngine(engineId) {
      if (!window.coveoHeadless[engineId].engine) {
        window.coveoHeadless[engineId].engine = initEngine(engineId);
      }
      return window.coveoHeadless[engineId].engine;
    }

Reply:
😅  yeah probably. ca fonctionne quand meme comme tu t'attend a une promesse anyway mais je vais l'ajouter

Reply:
Reading the code again, it feels weird to have an `async` function not `await`ing for something. But, I think it was correct since `window.coveoHeadless[engineId].engine` is meant to be a promise.

However, it means that the `await` is missing when we pass the engine to the `initializedCallback`.

    const executeInitializedCallback = (engineId) => {
      window.coveoHeadless[engineId].initializedCallback(window.coveoHeadless[engineId].engine);
    };

If my understanding is correct, we should really pass the engine itself to the callback, not a promise.

Reply:
yeah correct. sorry i'm waiting on sami's PR for dynamic reducer loading so i've neglected this one a bit and made a sloppy update last time

Reply:
No problem!